app/components/codelist-form.ts

```diff
@@ -28,6 +29,16 @@ export default class CodelistFormComponent extends Component<Args> {
   @tracked codelistTypes?: ArrayProxy<SkosConcept>;
   @tracked selectedType?: SkosConcept;
 
+  @tracked selectedIcon: IconModel | null = null;
+
+  get valueOptions() {
+    return this.options?.filter((model) => !(model instanceof IconModel));
+  }
+
+  get iconOptions() {
+    return this.options?.filter((model) => model instanceof IconModel);
+  }
```

Comment:
**Note:** The data split is handled in the component; the rest of the app manages icons as `skosConcept` (the superclass).